openshift_cli_installer/libs/managed_clusters/osd_clusters.py

```diff
@@ -62,15 +64,22 @@ def osd_create_cluster(cluster_data):
         set_cluster_auth(cluster_data=cluster_data, cluster_object=cluster_object)
 
         click.secho(
-            f"Cluster {cluster_data['name']} created successfully", fg=SUCCESS_LOG_COLOR
+            f"Cluster {cluster_name} created successfully", fg=SUCCESS_LOG_COLOR
         )
 
     except Exception as ex:
         click.secho(
-            f"Failed to run cluster create for cluster {cluster_data['name']}\n{ex}",
+            f"Failed to run cluster create for cluster {cluster_name}\n{ex}",
             fg=ERROR_LOG_COLOR,
         )
 
+        if must_gather_output_dir:
+            collect_must_gather(
+                must_gather_output_dir=must_gather_output_dir,
+                cluster_data=cluster_data,
+                cluster_object=cluster_object,
```

Comment:
don't we already have `cluster_object` in `cluster_data`?

Reply:
yes; moved the addition of it to earlier in the flow (till now it was after cluster create)